lms/djangoapps/mobile_api/video_outlines/serializers.py

```diff
@@ -48,23 +48,37 @@ def path(block):
             return reversed(block_path)
 
         def find_urls(block):
-            """section and unit urls for block"""
+            """
+            Find the section and unit urls for a block.
+
+            Returns:
+                unit_url, section_url:
+                    unit_url (str): The url of a unit
+                    section_url (str): The url of a section
+
+            """
             block_path = []
             while block in child_to_parent:
                 block = child_to_parent[block]
                 block_path.append(block)
 
-            course, chapter, section, unit = list(reversed(block_path))[:4]
-            position = 1
-            unit_name = unit.url_name
-            for block in section.children:
-                if block.name == unit_name:
-                    break
-                position += 1
+            block_list = list(reversed(block_path))
+            n = len(block_list)  # pylint: disable=C0103
+
+            chapter = block_list[1].location.block_id if n > 1 else None
+            section = block_list[2] if n > 2 else None
+            position = ""
+
+            if n > 3:
+                position = 1
+                for block in section.children:
+                    if block.name == block_list[3].url_name:
+                        break
+                    position += 1
 
             kwargs = dict(
-                course_id=course.id.to_deprecated_string(),
-                chapter=chapter.url_name,
+                course_id=self.course_id,
+                chapter=chapter,
```

Comment:
We don't use anything in the chapter block except for the url_name or location.block_id. I changed it back to just chapter to make it more consistent with the other blocks.